### DISCUSSION TITLE
Rule Request - Internal procedure with no references

### DISCUSSION THREAD

**Discussion by tscottjendev** [StefanMaron/BusinessCentral.LinterCop#460] [New Rule]
Not sure if this is something that fits your model but I would like to have a rule that checks to see if an internal procedure has no references.  I am currently refactoring and splitting a monolith app and find unreferenced code once I have made objects internal.

You would need to cover an `internal procedure` in an object with `Access = Public` and a `procedure` in an object with `Access = Internal`.

What do you think?

**Comment by Arthurvdv** [comment]
Good idea. I'm doubting on this rule in combination with the `internalsVisibleTo` property of the app.json.

The be honest we tent to use this for our testing apps, and using it more than I would admit 😅

When opening a multi-root workspace in VS Code this could work (although there's a challenge if the project isn't loaded). I'm unsure about this in a ADO/Github pipeline.

```json
  "internalsVisibleTo": [
    {
      "id": "51e7db57-be52-492d-a10c-ad26b45e255c",
      "publisher": "MyPublisher",
      "name": "MyApp.TestApp"
    }
  ],
```

**Comment by tscottjendev** [comment]
From perspective, the only use for `internalsVisibleTo` is for testing.  If you need it for a production app dependency scenario then this rule should be deprecated/ignored for the app.

Maybe there are two rules here.  One, a warning, if there is no `internalsVisibleTo`, and another an "information" if there is and make it the developers responsibility to determine if they need to act on it.

**Comment by rvanbekkum** [comment]
Duplicate of https://github.com/StefanMaron/BusinessCentral.LinterCop/discussions/197 ?

**Reply by Arthurvdv** [reply]
You're right, let's continue the discussion over there.
